service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantAboutToStartControllerTest.java

```diff
@@ -26,8 +56,94 @@ void shouldPrepopulateApplicantDataWhenNoApplicantExists() {
             .data(Map.of("data", "some data"))
             .build();
 
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken))
+            .willReturn(Organisation.builder().build());
+
         AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
 
         assertThat(callbackResponse.getData()).containsKey("applicants");
     }
+
+    @BeforeEach
+    void setup() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+    }
+
+    @Test
+    void shouldAddOrganisationDetailsToApplicantWhenOrganisationExists() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("data", "some data"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+
+        CaseData data = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(data.getAllApplicants()).contains(buildApplicant());
+    }
+
+    @Test
+    void shouldFindOrganisation() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+
+        Organisation actualOrganisation = organisationApi.findOrganisationById(userAuthToken, serviceAuthToken);
+
+        assertThat(actualOrganisation).isEqualTo(organisation);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenOrganisationNotFound() {
+        Exception exception = new FeignException.NotFound("", REQUEST, new byte[]{});
+        when(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).thenThrow(exception);
+
+        assertThatThrownBy(() -> organisationApi.findOrganisationById(userAuthToken, serviceAuthToken))
+            .isInstanceOf(FeignException.class);
+
+    }
+
+    private Organisation buildOrganisation() {
+        return Organisation.builder().name("Organisation")
+        .contactInformation(buildOrganisationContactInformation())
+            .build();
```

Comment:
is the spacing correct here? Maybe we should space down .name()

Reply:
changed :)